v2/containeranalysis/client.go

```diff
@@ -159,38 +159,40 @@ func (g *Client) Close() {
 // GetBuildDetail gets the BuildDetail for the passed image.
 func (g *Client) GetBuildDetail(ctx context.Context, ref reference.Canonical) (repository.BuildDetail, error) {
 	var err error
-	var parent string
-
-	filterStr := kindFilterStr(ref, grafeas.NoteKind_BUILD)
 
 	project, err := uri.ReferenceToProjectName(ref)
-	if nil != err {
+	if err != nil {
 		return repository.BuildDetail{}, err
 	}
 
-	appEnv := os.Getenv("APP_ENV")
-
-	if appEnv == "production" {
-		parent = projectPath(project)
-	} else {
-		parent = "projects/shopify-docker-images"
+	buildDetail, err := g.getBuildDetailFromProject(ctx, project, ref)
+	if err != nil && g.buildDetailProject != "" {
```

Comment:
```suggestion
	if err != nil && g.buildDetailProject != "" && g.buildDetailProject != project {
```

Nit: we can avoid the second check if the fallback project is the same as `project`.